api/trace/api.go

```diff
@@ -58,17 +58,30 @@ type (
 		Tracer() Tracer
 
 		Finish()
+
+		// IsRecordingEvents returns true is the span is active and recording events is enabled.
+		IsRecordingEvents() bool
+
+		// SpancContext returns span context of the span. Return SpanContext is usable
+		// even after the span is finished.
+		SpanContext() core.SpanContext
+
+		SetStatus(core.Status)
 	}
 
 	Injector interface {
 		Inject(core.SpanContext, tag.Map)
 	}
 
-	Option struct {
-		attribute  core.KeyValue
-		attributes []core.KeyValue
-		startTime  time.Time
-		reference  Reference
+	// Option apply changes to SpanOptions.
+    Option func(*SpanOptions)
+
+	SpanOptions struct {
```

Comment:
Is there a reason this is exported && the code uses functional config options?

Reply:
SDK would need this. Currently SDK is part of the same package but the next step is to separate it out.